pkg/cache/v2/cache.go

```diff
@@ -94,7 +126,29 @@ type RawResponse struct {
 	marshaledResponse atomic.Value
 }
 
+// RawDeltaResponse is a pre-serialized xDS response that utilizes the delta discovery request/response objects.
+type RawDeltaResponse struct {
+	// Request is the latest delta request on the stream.
+	DeltaRequest *discovery.DeltaDiscoveryRequest
+
+	// SystemVersionInfo is present for holding the current applied response version but generally used debugging purposes only.
+	SystemVersionInfo string
+
+	// Resources to be included in the response.
+	Resources []types.Resource
+
+	// RemovedResources is a list of resource aliases which should be dropped by the client consuming.
+	RemovedResources []string
+
+	// VersionMap is a map of versions applied internally to the cache grouped by resource name.
+	VersionMap map[string]string
```

Comment:
Same rename here to NextVersionMap?

Reply:
I think if we call this `NextVersionMap` we'll get a naming collision, so I'd have to change the method to maybe `GetNextVersionMap()`?

Reply:
I think whichever way we go it would be good be consistent with the naming